currency/storage.go

```diff
@@ -357,45 +377,66 @@ func (s *Storage) WriteCurrencyDataToFile(path string, mainUpdate bool) error {
 // LoadFileCurrencyData loads currencies into the currency codes
 func (s *Storage) LoadFileCurrencyData(f *File) error {
 	for i := range f.Contracts {
-		contract := f.Contracts[i]
-		contract.Role = Contract
-		err := s.currencyCodes.LoadItem(&contract)
+		if f.Contracts[i].Role != Contract {
+			return fmt.Errorf("%w %s expecting: %s",
+				errUnexpectedRole, f.Contracts[i].Role, Contract)
+		}
```

Comment:
While very difficult to get to this line, when we load data from the `File`, it does not have a `Role` set, so this will always fail, same with the other if statements in this function.

This is because in `currency.Item` you set the following:
```
	Role       Role   `json:"-"`
```
So no role can be saved in the way you are checking